languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10084,6 +10084,378 @@ though, and has since been slightly modified)-->
+              <example correction="grand, cela sera">Le plus <marker>grand cela sera</marker> Pierre.</example>
+              <example correction="fort, cela serait">Le plus <marker>fort cela serait</marker> Christophe.</example>
+              <example>Il sera grand <marker>et ce sera</marker> un homme.</example>
+              <example>Il m'a dit <marker>que ce serait</marker> un homme.</example>
+              <example>Il se demande <marker>comment cela est</marker> possible.</example>
+              <example>Ce film a été réalisé avec cet homme, <marker>dont ce sera</marker> le dernier film.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token regexp="yes">quelle|quel|quelles|quels|comment|où|pourquoi</token>
+                  <token regexp="yes">[^,);:(«»-]+</token>
+                  <token regexp="yes">cela|ça</token>
+                  <token postag="V etre.* 3 s" postag_regexp="yes"/>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">[^,);:(«»-]+<exception postag="C sub|C coor|R rel e sp|A inte|P" postag_regexp="yes"/><exception regexp="yes">mieux|car|travers|alors|même|si|tout|sinon|en|plus|comment|où|pourquoi</exception></token>
+                      <token regexp="yes">ça|cela</token>
+                      <token postag="V avoir.* 3 s" postag_regexp="yes"/>
+                      <token postag="V etre ppa.*" postag_regexp="yes"/>
+                  </marker>
+              </pattern>
+              <message>Une virgule est requise dans ce contexte.</message>
+              <suggestion>\1, \2 \3 \4</suggestion>
+              <example correction="fort, ça aurait été">Le plus <marker>fort ça aurait été</marker> Christophe.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token postag="SENT_START"/>
+                  <token regexp="yes" min="1" max="3">[^,);:(«»-]+</token>
+                  <token regexp="yes">mais|car</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes">[^,);:(«»-]+</token>
+                  <token regexp="yes">mais|car</token>
+                  <token regexp="yes" min="1" max="3">[^,);:(«»-]+</token>
+                  <token postag="SENT_END"/>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">[^,);:(«»-]+</token>
+                      <token regexp="yes">mais|car</token>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="fâchée, mais">Depuis qu'il me l'a dit, je suis <marker>fâchée mais</marker> je ne lui en veux pas.</example>
+              <example>Il est <marker>beau mais</marker> grand.</example>
+          </rule>
+          <rule>
```

Comment:
This subrule is also very nice. I like the idea of creating separate subrules for those cases. Here, you'll just have to take care of the quotation marks, like everywhere, and check again for FPs like this:

> Certains s'y allièrent aux familles de gentilshommes locaux — voire bourgeoises — en donnant naissance à de nouvelles lignées.

Hint: you can write the n-dash (–) with ALT + 0150 and the m-dash (—) with ALT + 0151 (numbers to be typed on the numpad)